pkg/resources/manager/manager.go

```diff
@@ -360,18 +398,22 @@ func (r *resourceManager) getAvailableMachineClasses() []*iri.MachineClassStatus
 	return status
 }
 
-func (r *resourceManager) calculateMachineClassQuantity(class *machineclass) error {
+func (r *resourceManager) calculateMachineClassQuantity(class *MachineClass) error {
 	var count int64 = math.MaxInt64
 
-	for key, classQuantity := range class.resources {
-		available, err := getResource(key, r.getAvailableResources())
-		if err != nil {
-			return err
+	for _, s := range r.sources {
```

Comment:
Can't we use here the same mechanism that is being used above s, ok := r.registredResources[key] , in place of iterating over all the sources?

Reply:
fixed